src/cmd/linuxkit/run_qemu.go

```diff
@@ -458,8 +472,10 @@ func buildQemuCmdline(config QemuConfig) (QemuConfig, []string) {
 	qemuArgs = append(qemuArgs, "-pidfile", filepath.Join(config.StatePath, "qemu.pid"))
 	// Need to specify the vcpu type when running qemu on arm64 platform, for security reason,
 	// the vcpu should be "host" instead of other names such as "cortex-a53"...
-	if config.Arch == "aarch64" {
+	if config.Arch == "aarch64" && runtime.GOARCH == "arm64" {
 		qemuArgs = append(qemuArgs, "-cpu", "host")
+	} else {
+		qemuArgs = append(qemuArgs, "-cpu", "cortex-a57")
```

Comment:
This will try and use cortex-a57 even if you are running on (and _want_) an x86 won't it?